### PR TITLE
network: several CAN device related updates

### PR DESCRIPTION
**Pull request by yuwata**
Replaces #19471.

### CODE REVIEW

**Comment by yuwata** [comment]
@clemensg Could you test this PR?

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -2567,6 +2575,9 @@ static int link_carrier_gained(Link *link) {
 
         assert(link);
 
+        if (link->iftype == ARPHRD_CAN)
+                return link_handle_bound_by_list(link);
```
Comment (poettering) [inline src/network/networkd-link.c:2591]: maybe add comment here, `/* let's shortcut things for CAN which doesn't need most of what#s done below. */` or something like that?

**Comment by poettering** [comment]
i have not much clue about CAN, but changes look superficially ok to me

**Comment by clemensg** [comment]
> @clemensg Could you test this PR?

@yuwata I tested it on an embedded board with two CAN interfaces (flexcan). Everything still working, no more error message and the state is now "carrier (configured)". 👍 

I will close the other PR.

**Comment by yuwata** [comment]
@poettering Thank you for review.
@clemensg Thank you for testing.

The suggested comment is added. Upgrading the green label.

**Pull request merged**